src/server/editorServices.ts

```diff
@@ -1388,6 +1397,7 @@ namespace ts.server {
                 this.documentRegistry,
                 compilerOptions,
                 /*lastFileExceededProgramSize*/ this.getFilenameForExceededTotalSizeLimitForNonTsFiles(projectFileName, compilerOptions, files, externalFilePropertyReader),
+                /*languageServicePermanentlyDisabled*/this.syntaxOnly,

```

Comment:
why do you need to pass this to each project, that should be checked by each project by just checking this.projectService.syntaxOnly?

Reply:
Just an oversight.